src/evaluator/function/arithmetic.cpp

```diff
@@ -0,0 +1,78 @@
+#include "../../ast/kind.h"
+#include "../error.h"
+#include "../function.h"
+#include <functional>
+#include <memory>
+#include <optional>
+#include <string>
+
+namespace evaluator {
+
+using ast::Element;
+using ast::ElementKind;
+
+std::optional<double> cast_to_double(std::shared_ptr<Element> element) {
+    if (auto integer = std::dynamic_pointer_cast<ast::Integer>(element)) {
+        return integer->value;
+    } else if (auto real = std::dynamic_pointer_cast<ast::Real>(element)) {
+        return real->value;
+    } else {
+        return std::nullopt;
+    }
+}
+
+std::shared_ptr<Element> evaluate_arithmetic_operation(
+    CallFrame const& frame,
+    std::function<double(int, int)> operation_int,
+    std::function<double(double, double)> operation_double
+) {
+    auto a_element = frame.arguments[0];
+    auto b_element = frame.arguments[1];
+    if (a_element->kind == ElementKind::INTEGER &&
+        b_element->kind == ElementKind::INTEGER) {
+        auto a_integer = std::dynamic_pointer_cast<ast::Integer>(a_element);
+        auto b_integer = std::dynamic_pointer_cast<ast::Integer>(b_element);
+        return std::make_shared<ast::Integer>(
+            operation_int(a_integer->value, b_integer->value), frame.call_site
+        );
+    }
```

Comment:
```suggestion
    if (auto a_integer = std::dynamic_pointer_cast<ast::Integer>(a_element) &&
        auto b_integer = std::dynamic_pointer_cast<ast::Integer>(b_element)) {
        return std::make_shared<ast::Integer>(
            operation_int(a_integer->value, b_integer->value), frame.call_site
        );
    }
```

или плюсы так не могут?

Reply:
не могут

Reply:
я не нашел как это норм написать и сделаем по простому